core/src/apps/cardano/sign_tx/pool_owner_signer.py

```diff
@@ -27,23 +27,25 @@ def __init__(
     ) -> None:
         super().__init__(ctx, msg, keychain)
 
+        self.signing_mode_title = "Confirming a transaction as pool owner."
+
     def _validate_tx_init(self) -> None:
         super()._validate_tx_init()
-        if (
-            self.msg.certificates_count != 1
-            or self.msg.withdrawals_count != 0
-            or self.msg.minting_asset_groups_count != 0
-        ):
-            raise wire.ProcessError(
-                "Stakepool registration transaction cannot contain other certificates, withdrawals or minting"
-            )
+        self._assert_tx_init_cond(self.msg.certificates_count == 1)
+        self._assert_tx_init_cond(self.msg.withdrawals_count == 0)
+        self._assert_tx_init_cond(self.msg.minting_asset_groups_count == 0)
+        self._assert_tx_init_cond(self.msg.script_data_hash is None)
+        self._assert_tx_init_cond(self.msg.collateral_inputs_count == 0)
+        self._assert_tx_init_cond(self.msg.required_signers_count == 0)
+        self._assert_tx_init_cond(not self.msg.has_collateral_return)
+        self._assert_tx_init_cond(self.msg.total_collateral is None)
+        self._assert_tx_init_cond(self.msg.reference_inputs_count == 0)
 
-        if (
-            self.msg.script_data_hash is not None
-            or self.msg.collateral_inputs_count != 0
-            or self.msg.required_signers_count != 0
-        ):
-            raise wire.ProcessError("Invalid tx signing request")
+    def _should_show_tx_init(self) -> bool:
+        return False
+
+    def _is_expert_view_allowed(self) -> bool:
+        return False
```

Comment:
Does this mean that if a pool registration tx contains an auxiliary data hash, there is no way for the user to view it?

Reply:
Indeed that was the case 🙈 https://github.com/vacuumlabs/trezor-firmware/pull/79/commits/eddc71d2c1978a113e8acefe7ac1f7c77d413067